HandleNHNotifyMessageCallOrchestrator/__tests__/handler.test.ts

```diff
@@ -0,0 +1,134 @@
+} from "../../HandleNHNotifyMessageCallActivity/handler";
+
+import { getHandler, NhNotifyMessageOrchestratorCallInput } from "../handler";
+
+import { envConfig } from "../../__mocks__/env-config.mock";
+import {
+  callableActivity,
+  OrchestratorFailure,
+  OrchestratorSuccess,
+  success as orchestratorSuccess
+} from "../../utils/durable/orchestrators";
+import { IOrchestrationFunctionContext } from "durable-functions/lib/src/iorchestrationfunctioncontext";
+import { NotificationHubConfig } from "../../utils/notificationhubServicePartition";
+
+const aNotificationHubConfig: NotificationHubConfig = {
+  AZURE_NH_ENDPOINT: envConfig.AZURE_NH_ENDPOINT,
+  AZURE_NH_HUB_NAME: envConfig.AZURE_NH_HUB_NAME
+};
+
+const aFiscalCodeHash = "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855" as NonEmptyString;
+
+const aNotifyMessage: NotifyMessage = {
+  installationId: aFiscalCodeHash,
+  kind: NotifyMessageKind.Notify,
+  payload: {
+    message: "message",
+    message_id: "id",
+    title: "title"
+  }
+};
+
+const nhCallOrchestratorInput = NhNotifyMessageOrchestratorCallInput.encode({
+  message: aNotifyMessage
+});
+
+const retryOptions = {
+  ...new df.RetryOptions(5000, envConfig.RETRY_ATTEMPT_NUMBER),
+  backoffCoefficient: 1.5
+};
+
+const notifyMessageActivity = callableActivity<NotifyMessageActivityBodyImpl>(
+  "HandleNHNotifyMessageCallActivity",
+  NotifyMessageActivityResultSuccess,
+  retryOptions
+);
+
+const mockGetInput = jest.fn<unknown, []>(() => nhCallOrchestratorInput);
+const contextMockWithDf = ({
+  ...contextMock,
+  df: {
+    callActivityWithRetry: jest.fn().mockReturnValue(activitySuccess()),
+    getInput: mockGetInput
+  }
+} as unknown) as IOrchestrationFunctionContext;
+
+describe("HandleNHNotifyMessageCallOrchestrator", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("should start the activities with the right inputs", async () => {
+    const orchestratorHandler = getHandler({
+      notifyMessageActivity,
+      legacyNotificationHubConfig: aNotificationHubConfig
+    })(contextMockWithDf as any);
+
+    // call orchestrator 1 time
+    orchestratorHandler.next();
+
+    expect(contextMockWithDf.df.callActivityWithRetry).toBeCalledWith(
+      "HandleNHNotifyMessageCallActivity",
+      retryOptions,
+      NHCallServiceActivityInput.encode({
+        message: aNotifyMessage,
+        notificationHubConfig: aNotificationHubConfig
+      })
+    );
+  });
+
+  it("should end the activity with SUCCESS in two steps", async () => {
+    const orchestratorHandler = getHandler({
+      notifyMessageActivity,
+      legacyNotificationHubConfig: aNotificationHubConfig
+    })(contextMockWithDf as any);
+
+    // call orchestrator 1 time
+    const res = orchestratorHandler.next();
+    expect(res.done).toBeFalsy();
+
+    // call orchestrator 2nd time (expected to be the last one)
+    const res2 = orchestratorHandler.next(res.value as any);
+    expect(res2.done).toBeTruthy();
```

Comment:
You are actually testing the JS implementation of a Generator function 😉 .

We should only care on the final result of the generator, and it should not be known how many _steps_ are needed for the orchestrator to complete (which are not fixed numbers, imagine an orchestrator with different branches).

We have the `consumeGenerator` helper that you may find useful. See an example in https://github.com/pagopa/io-functions-pushnotifications/blob/master/HandleNHCreateOrUpdateInstallationCallOrchestrator/__tests__/handler.test.ts#L95

Reply:
I saw it, really helpful!
Yes, you're right, there's no added value in this